Apromore-Custom-Plugins/Process-Simulation-Info-Logic/src/main/java/org/apromore/processsimulation/service/SimulationInfoService.java

```diff
@@ -277,7 +305,7 @@ private void deriveTimetable(
                         .id(resourceId)
                         .name(resourceName)
                         .totalAmount(roleToResourceCount.getValue())
-                        .timetableId(config.getDefaultTimetable().get(CONFIG_DEFAULT_ID_KEY))
+                        .timetableId(config.getTimetable().get(CONFIG_CUSTOM_ID_KEY))
```

Comment:
config.getTimetable().get(CONFIG_CUSTOM_ID_KEY): this can be null if the timetable map is not initialized with CONFIG_CUSTOM_ID_KEY, depending on the values put in the properties file (see SimulationInfoConfig), which is not safe.

Reply:
The custom ID is part of the timetable properties map and the configuration is initialised at the time it gets autowired.

```
process-simulation-info-export.timetable.id=DEFAULT_TIMETABLE
process-simulation-info-export.timetable.custom-id=CUSTOM_TIMETABLE
process-simulation-info-export.timetable.timeslot-name=Default Timeslot
```

Reply:
I see the need of making some names configurable via .properties file, e.g. "CUSTOM_TIMETABLE", "Default Timeslot". We're mixing both properties reader object and config object in one SimulationInfoConfig. One way of doing this without the above issue is create a separate properties reader object via Spring annotation, read properties values into that object, then pass that object to SimulationInfoConfig, where its getter reads the properties and returns the non-empty property value or otherwise returns a private constant. No map is used.

Reply:
I have made the change to this to simplify the loading of properties, and avoid having to use maps. Will address the immutability of the class in a later PR

Reply:
Ok.